src/h5web/providers/h5core/h5core-api.ts

```diff
@@ -0,0 +1,151 @@
+import {
+  ValueRequestParams,
+  Attribute,
+  Entity,
+  Dataset,
+  EntityKind,
+  Group,
+} from '../models';
+import { isDatasetResponse, isGroupResponse } from './utils';
+import { assertDataset } from '../../guards';
+import { ProviderApi } from '../api';
+import type {
+  H5CoreDataResponse,
+  H5CoreAttrResponse,
+  H5CoreMetaResponse,
+  H5CoreDatasetMetaReponse,
+  H5CoreGroupMetaResponse,
+} from './models';
+import { convertDtype, flattenValue } from '../utils';
+
+export class H5CoreApi extends ProviderApi {
+  /* API compatible with h5core@c6093e407d9ba20cec586d29232a9bbcb490835e */
+  public constructor(url: string, filepath: string) {
+    super(filepath, { baseURL: url });
+  }
+
+  public async getEntity(path: string): Promise<Entity> {
+    return this.processEntity(path, 1);
+  }
+
+  public async getValue(params: ValueRequestParams): Promise<unknown> {
+    const { path, selection } = params;
+    const [value, entity] = await Promise.all([
+      this.fetchData(params),
+      this.getEntity(path),
+    ]);
+
+    assertDataset(entity);
+
+    return flattenValue(value, entity, selection);
+  }
+
+  protected async fetchAttributes(path: string): Promise<H5CoreAttrResponse> {
+    const { data } = await this.client.get<H5CoreAttrResponse>(
+      `/attr/${this.filepath}?path=${path}`
+    );
+    return data;
+  }
+
+  protected async fetchData(
+    params: ValueRequestParams
+  ): Promise<H5CoreDataResponse> {
+    const { path, selection = '' } = params;
+    const { data } = await this.cancellableFetchValue<H5CoreDataResponse>(
+      `/data/${this.filepath}?path=${path}${
+        selection && `&selection=${selection}`
+      }`,
+      params
+    );
+    return data;
+  }
+
+  protected async fetchMetadata(path: string): Promise<H5CoreMetaResponse> {
+    const { data } = await this.client.get<H5CoreMetaResponse>(
+      `/meta/${this.filepath}?path=${path}`
+    );
+    return data;
+  }
```

Comment:
The fetch functions are the same as in `JupyterApi` except that the url were adapted to h5core and that we don't have a content endpoint in h5core (`children` are returned by the meta endpoint)

Reply:
nice ^^